src/types/infer.ts

```diff
@@ -2341,6 +2343,48 @@ export class InferType {
         return false;
     }
 
+    isABIEncodable(type: TypeNode, encoderVersion: ABIEncoderVersion): boolean {
```

Comment:
If we add some new builtin type, we might forget to add it to this `if` and `isABIEncodable` would automatically assume that that type is encodable. I would prefer if we flip the logic, that we `isABIEncodable` explicitly lists the allowed types, and by default it returns `false` at the end.

Reply:
Addressed in c4c55d6